sw/device/lib/aes.c

```diff
@@ -0,0 +1,100 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/aes.h"
+
+#include "aes_regs.h"  // Generated.
+#include "sw/device/lib/common.h"
+
+#define AES0_BASE_ADDR 0x40110000
+
+void aes_init(aes_cfg_t aes_cfg) {
+  // Convert key_len from bytes to onehot
+  size_t key_len_onehot = 0b001;
+  if (aes_cfg.key_len_in_bytes == 24) {
+    key_len_onehot = 0b010;
+  } else if (aes_cfg.key_len_in_bytes == 32) {
+    key_len_onehot = 0b100;
+  }
+
+  REG32(AES_CTRL(0)) =
```

Comment:
As long as there is exactly one "operation"/bit-field set per line, I do not really prefer one over the other. In case, the formatter would decide to have multiple bit-fields set per line, your suggestion does for help to keep the code readable. In this particular case, I did not change the code.